editoast/src/views/timetable/stdcm.rs

```diff
@@ -220,107 +157,56 @@ async fn stdcm(
         })
         .await?;
 
-    let towed_rolling_stock =
-        if let Some(towed_rolling_stock_id) = stdcm_request.towed_rolling_stock_id {
-            let towed_rolling_stock =
-                TowedRollingStockModel::retrieve_or_fail(conn, towed_rolling_stock_id, || {
-                    STDCMError::TowedRollingStockNotFound {
-                        towed_rolling_stock_id,
-                    }
-                })
-                .await?;
-            Some(towed_rolling_stock)
-        } else {
-            None
-        };
-
-    let simulations = train_simulation_batch(
-        conn,
-        valkey_client.clone(),
-        core_client.clone(),
-        &train_schedules,
-        &infra,
-        stdcm_request.electrical_profile_set_id,
-    )
-    .await?;
+    // 2. Compute the earliest start time and maximum departure delay
+    let (virtual_train_schedule, virtual_train_sim_result, virtual_train_pathfinding_result) =
+        simulate_train_run(
+            db_pool.clone(),
+            valkey_client.clone(),
+            core_client.clone(),
+            &stdcm_request,
+            &infra,
+            &rolling_stock,
+            timetable_id,
+        )
+        .await?;
 
-    // 2. Compute the earliest start time, maximum running time and maximum departure delay
-    // Simulation time without stop duration
-    let (
-        simulation_run_time,
-        virtual_train_schedule,
-        virtual_train_sim_result,
-        virtual_train_pathfinding_result,
-    ) = simulate_train_run(
-        db_pool.clone(),
-        valkey_client.clone(),
-        core_client.clone(),
-        &stdcm_request,
-        &infra,
-        &rolling_stock,
-        timetable_id,
-    )
-    .await?;
-    let simulation_run_time = match simulation_run_time {
-        SimulationTimeResult::SimulationTime { value } => value,
-        SimulationTimeResult::Error { error } => {
+    let simulation_run_time = match virtual_train_sim_result.clone().simulation_run_time() {
```

Comment:
```suggestion
    let simulation_run_time = match virtual_train_sim_result.simulation_run_time() {
```

The syntax `let Ok(..) = .. else ` covers this precise pattern ;)

Reply:
Done.